loopy/transform/privatize.py

```diff
@@ -246,4 +248,137 @@ def privatize_temporaries_with_inames(
 # }}}
 
 
+# {{{ unprivatize temporaries with iname
+
+class _InameRemover(IdentityMapper):
+    def __init__(self, inames_to_remove, only_var_names):
+        self.only_var_names = only_var_names
+        self.inames_to_remove = inames_to_remove
+        self.var_name_to_remove_indices = {}
+        super().__init__()
+
+    def map_subscript(self, expr, in_subscript=False):
+        name = expr.aggregate.name
+        if not self.only_var_names or name in self.only_var_names:
+            index = expr.index
+            if not isinstance(index, tuple):
+                index = (index,)
+
+            remove_indices = {}
+            new_index = []
+            for i, index_expr in enumerate(index):
+                if isinstance(index_expr, pymbolic.primitives.Variable) and \
+                        index_expr.name in self.inames_to_remove:
+                    remove_indices[i] = index_expr.name
+                else:
+                    new_index.append(index_expr)
+
+            if name in self.var_name_to_remove_indices:
+                old_remove_indices = self.var_name_to_remove_indices[name]
+                if old_remove_indices != remove_indices:
+                    raise LoopyError(f"Cannot remove indices {remove_indices}"
+                         f" for subscript {expr} because there was another"
+                         f" subscript that required removing indices"
+                         f" {old_remove_indices}")
```

Comment:
I'm not sure I understand this error message. Could you work on the phrasing?

Reply:
I tried in 0a7fe3ffbe8606008794a643bb18ffdc597097b6 with a slight improvement. Not sure if that's enough. This error message is about failing when there's something like `acc[i, j] = acc[j, i] + 1` and loopy is asked to unprivatize `j`.